test/e2e_node/container_lifecycle_test.go

```diff
@@ -398,6 +398,184 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
 		framework.ExpectNoError(results.ExitsBefore(prefixedName(PostStartPrefix, regular1), regular2))
 	})
 
+	ginkgo.When("A pod is running a regular container with restartPolicy=Never", func() {
```

Comment:
nit: For PodSpecs with `restartPolicy=Never` and `restartPolicy=OnFailure`, I think a single test case for the container image update is sufficient, so it might be fine to simply replace it with `ginkgo.It()`.